### ISSUE TITLE
asynccontextmanager reportArgumentType error on empty abstractmethod

### ISSUE THREAD

**Issue by sherbang** [microsoft/pyright#10178]
**Describe the bug**
When using asynccontextmanager on an abstractmethod, if the method has no body, then a type error is reported on the asynccontextmanager decorator line.

![Image](https://github.com/user-attachments/assets/0d7db2cc-cb8d-4b49-b96c-cf70975a0570)

**Code or Screenshots**
```python
from abc import ABCMeta, abstractmethod
from collections.abc import AsyncIterator
from contextlib import asynccontextmanager


class MyCaller(metaclass=ABCMeta):
    @abstractmethod
    @asynccontextmanager
    async def no_error(self) -> AsyncIterator[None]:
        yield

    @abstractmethod
    @asynccontextmanager
    async def error(self) -> AsyncIterator[None]: ...
```

Error:
```
pyright_contextmanager.py:13:6 - error: Argument of type "(self: Self@MyCaller) -> CoroutineType[Any, Any, AsyncIterator[None]]" cannot be assigned to parameter "func" of type "(**_P@asynccontextmanager) -> AsyncIterator[_T_co@asynccontextmanager]" in function "asynccontextmanager"
    Type "(self: Self@MyCaller) -> CoroutineType[Any, Any, AsyncIterator[None]]" is not assignable to type "(**_P@asynccontextmanager) -> AsyncIterator[_T_co@asynccontextmanager]"
      Function return type "CoroutineType[Any, Any, AsyncIterator[None]]" is incompatible with type "AsyncIterator[_T_co@asynccontextmanager]"
        "CoroutineType[Any, Any, AsyncIterator[None]]" is incompatible with protocol "AsyncIterator[_T_co@asynccontextmanager]"
          "__anext__" is not present
          "__aiter__" is not present (reportArgumentType)
1 error, 0 warnings, 0 informations 
```

**VS Code extension or command-line**
Reproduced with pyright command line versions 1.1.395 through 1.1.398 and Pylance 2025.3.2

**Comment by erictraut** [comment]
Yes, this behavior is correct. See [this issue](https://github.com/microsoft/pyright/issues/9949) for an explanation.
